skimage/draw/draw.py

```diff
@@ -107,22 +108,33 @@ def ellipse(r, c, r_radius, c_radius, shape=None, rotation=0.):
            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
            [1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 1]], dtype=uint8)
     """
+    # <-- Starting deprecation warnings: rotation -> orientation,
+    # shape <-> orientation.
+    if rotation is not None:
+        warn('The keyword rotation will be replaced with orientation in version '
+             '0.19.', DeprecationWarning, stacklevel=2)
+        orientation = rotation
+    if shape is not None or not isinstance(shape, tuple):
+        warn('The order of the keywords shape and orientation will be changed '
+             'in version 0.19.', FutureWarning, stacklevel=2)
+    # <-- Ending deprecation warnings: rotation -> orientation,
+    # shape <-> orientation.
 
     center = np.array([r, c])
     radii = np.array([r_radius, c_radius])
-    # allow just rotation with in range +/- 180 degree
-    rotation %= np.pi
-
-    # compute rotated radii by given rotation
-    r_radius_rot = abs(r_radius * np.cos(rotation)) \
-                   + c_radius * np.sin(rotation)
-    c_radius_rot = r_radius * np.sin(rotation) \
-                   + abs(c_radius * np.cos(rotation))
+    # allow orientation only within range +/- pi
+    orientation %= np.pi
```

Comment:
Why does this line even exist in the first place

Reply:
Not sure. Maybe the first implementer wanted to restrict the interval.
My vote is to remove it, keeping it simpler.